pdfgen.c

```diff
@@ -2378,15 +2380,38 @@ static pdf_object *pdf_add_raw_jpeg_data(struct pdf_doc *pdf,
                 "/BitsPerComponent 8\r\n/Filter /DCTDecode\r\n"
                 "/Length %d\r\n>>stream\r\n",
                 flexarray_size(&pdf->objects),
-                ncolours == 1 ? "/DeviceGray" : "/DeviceRGB", width, height,
-                (int)len);
+                ncolours == 1 ? "/DeviceGray" : "/DeviceRGB", *width_read,
+                *height_read, (int)len);
     dstr_append_data(&obj->stream, jpeg_data, len);
 
     dstr_printf(&obj->stream, "\r\nendstream\r\n");
 
     return obj;
 }
 
+/**
+ * Get the display dimensions of an image, respecting the images aspect ratio
+ * if only one desired display dimension is defined.
+ */
+static void get_img_display_dimensions(uint32_t img_width,
```

Comment:
Can we make this return a bool, so we can indicate a failure (ie: img_width == img_height == 0, or display_width < 0 && display_height < 0)? I see your comment about it already checked in pdf_add_image_data, but I think there are other failure modes. div by zero for example.

Reply:
Yeah that sounds good.
If this fails however, what error value should I set in the PDF?
I guess `EINVAL` is the closest fit since it indicates "Invalid argument".

Reply:
Yes, `-EINVAL` is probably the best. That's sort of the catch-all I use almost everywhere whenever initialisation things go wrong.